src/net/connect.cpp

```diff
@@ -18,12 +19,9 @@ void mk_bufferevent_on_event(bufferevent *bev, short what, void *ptr) {
     if ((what & BEV_EVENT_CONNECTED) != 0) {
         (*cb)(mk::NoError(), bev);
     } else if ((what & BEV_EVENT_TIMEOUT) != 0) {
-        bufferevent_free(bev);
-        (*cb)(mk::net::TimeoutError(), nullptr);
+        (*cb)(mk::net::TimeoutError(), bev);
     } else {
-        bufferevent_free(bev);
-        // TODO: here we should map to the actual error that occurred
```

Comment:
Yes you are right and no I don't think it's wise to change the constructor of `NetworkError` as it would make it not aligned to the constructors of the other `Error` subclasses.

I will put this comment back then.